test-coverage-enforcer/src/main/java/org/wso2/testcoverageenforcer/Maven/POM/ParentPom.java

```diff
@@ -0,0 +1,143 @@
+            throws PomFileReadException, PomFileWriteException {
+
+        Document pomFile = DocumentReader.readDocument(pomFilePath);
+        pomFile.setDocumentURI(pomFilePath);
+        HashMap<String, Object> processedData = JacocoCoverage.insertJacocoCoverageCheck(
+                pomFile,
+                Constants.Maven.MAVEN_TAG_PLUGIN_MANAGEMENT,
+                coveragePerElement,
+                coverageThreshold);
+        Document jacocoInsertedPom = (Document) processedData.get(Constants.Jacoco.JACOCO_INSERTED_POM);
+        DocumentWriter.writeDocument(jacocoInsertedPom, pomFilePath);
+        return processedData;
+    }
+
+    /**
+     * Deep traverse through children and inherit coverage check from the parent
+     *
+     * @param children             A list of child maven objects
+     * @param coveragePerElement   Per which element jacoco coverage check should be performed
+     * @param coverageThreshold    Line coverage threshold to break the build
+     * @param surefireArgumentLine surefire argument name in the parent pom
+     * @param jacocoReportPath     jacoco report file path used in the parent pom
+     * @return Check rule added at least in any child
+     * @throws PomFileReadException  Error while reading the pom
+     * @throws PomFileWriteException Error while writing the pom file
+     */
+    public boolean inheritCoverageCheckInChildren(List<ChildPom> children, String coveragePerElement, String coverageThreshold,
+                                                  String surefireArgumentLine, String jacocoReportPath)
+            throws PomFileReadException, PomFileWriteException {
+
+        boolean checkRuleAddition = false; // Jacoco coverage check rule added at least in any child
+        for (ChildPom child : children) {
+            if (child.hasChildren()) {
+                boolean ruleAdded = inheritCoverageCheckInChildren(child.getChildren(), coveragePerElement, coverageThreshold, surefireArgumentLine, jacocoReportPath);
+                checkRuleAddition = checkRuleAddition || ruleAdded;
+            } else if (child.hasTests()) {
+                child.inheritCoverageCheckFromParent(coveragePerElement, coverageThreshold, surefireArgumentLine, jacocoReportPath);
+                checkRuleAddition = true;
+            } else if (!child.hasTests()) {
+                log.debug("Ignoring child module due to missing tests in " + child.getPomFilePath());
+            }
+
+        }
+        return checkRuleAddition;
+    }
+
+    /**
+     * Get minimum line coverage value in the project
+     *
+     * @return Minimum coverage ratio value among bundles
+     * @throws IOException          Error in opening files
+     * @throws PomFileReadException Error while reading the pom
+     */
+    public double getMinimumBundleCoverage(List<ChildPom> children)
+            throws PomFileReadException, IOException {
```

Comment:
Formatting issue

Reply:
Fixed in 2bc0618a39bb056608ff116edf53bc2b3f4e8099.